### PR TITLE
Refactor repos

### PR DESCRIPTION
**Pull request by jpc20**
Refactor Github service and request to take a token as an argument so the repos on the user dashboard are correlated to the current user.

### CODE REVIEW

File: app/services/github_service.rb [hunk 1]
```diff
@@ -1,12 +1,12 @@
...
-    repos = conn.get("/user/repos")
+  def repos(token)
+    repos = conn(token).get('/user/repos')
     JSON.parse(repos.body, symbolize_names: true)
   end
 
-  def conn
-    Faraday.new("https://api.github.com") do |req|
-      req.headers["Authorization"] = "token #{ENV["GITHUB_TOKEN"]}"
+  def conn(token)
```
Comment (jpc20) [inline app/services/github_service.rb:9]: This allows us to pass the current users token as an argument so the repos shown reflect the current user.

File: spec/features/user/user_has_a_github_section_spec.rb [hunk 2]
```diff
@@ -19,17 +19,20 @@
...
+    user_2_repos = GithubResults.new.repos(user_2.token)
     allow_any_instance_of(ApplicationController).to receive(:current_user).and_return(user_1)
 
     visit(dashboard_path)
 
-    within('.github') do
-      expect()
+    within(first('.repo-link')) do
+      expect(page).to have_content(user_1_repos.first.name)
+      expect(page).to_not have_content(user_2_repos.first.name)
```
Comment (jpc20) [inline spec/features/user/user_has_a_github_section_spec.rb:33]: Test that the repos shown belong to the current user.

File: app/controllers/users_controller.rb [hunk 3]
```diff
@@ -1,8 +1,6 @@
 class UsersController < ApplicationController
   def show
-
-    github_results = GithubResults.new
-    @repos = github_results.repos
+    @repos = GithubResults.new.repos(current_user.token) if current_user.token
```
Comment (madhalle) [inline app/controllers/users_controller.rb:3]: nice! Dynamic and condensed

File: app/services/github_service.rb (hunk 1 above)
Comment (madhalle) [reply app/services/github_service.rb:9]: So this is basically what we skipped testing yesterday by specifying which repo gets displayed

File: spec/features/user/user_has_a_github_section_spec.rb (hunk 2 above)
Comment (madhalle) [reply spec/features/user/user_has_a_github_section_spec.rb:33]: awesome

**Comment by madhalle** [comment]
Nice & travis looks happy

**Pull request merged**